src/plugins/vis_builder/public/application/utils/state_management/redux_persistence.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { VisBuilderServices } from '../../../types';
+import { getPreloadedState } from './preload';
+import { RootState } from './store';
+
+export const loadReduxState = async (services: VisBuilderServices) => {
```

Comment:
ideally both of these functions should be covered by unit tests

Reply:
added unit tests for both functions.